### ISSUE TITLE
Inflector::pluralize cause problem with double irregular words

### ISSUE THREAD

**Issue by atmacola** [cakephp/cakephp#6879]
I am trying to update to 2.6.7 but I have a problem with SQL hasAndBelongsToMany relations.
2 Tables with irregular words, no more work because of change in Inflector https://github.com/cakephp/cakephp/commit/323e8d8d76af3d182b4c3ebe2ac3ae26f28d2b84#diff-97b47e5516d221eddf4268c7e8f079ae

If you haved tables's name with number at the end (for example), you may want the singular to be mytable1 and the plural mytables1. 
```
$irregularPlural = array(
  'mytable1' => 'mytables1',
  'mytable2' => 'mytables2',
  'mytable1_mytable2' => 'mytables1_mytables2'
);

$result = Inflector::tableize( 'Mytable1Mytable2' );
// before 2.6.x => $result = 'mytables1_mytables2';
// after => $result = 'mytable1_mytables2'; // (No "s" in mytable1)
```

**Comment by markstory** [comment]
Does it work if you put `mytable1_mytable2` before the single word versions?  Fixing this could be problematic as it could regress #6544

**Comment by atmacola** [comment]
No this do not work.
 I have made this little modification in lib/Cake/Utility/Inflector.php#389 :

```
if (preg_match('/(.*(?:\\b|_))(' . self::$_plural['cacheIrregular'] . ')$/i', $word, $regs)) {
	if (preg_match('/^' . self::$_plural['cacheIrregular'] . '$/i', $word, $matches)) {
		self::$_cache['pluralize'][$word] = substr($matches[0], 0, 1) . substr(self::$_plural['merged']['irregular'][strtolower($matches[0])], 1);
	} else {
		self::$_cache['pluralize'][$word] = $regs[1] . substr($regs[2], 0, 1) . substr(self::$_plural['merged']['irregular'][strtolower($regs[2])], 1);
	}

	return self::$_cache['pluralize'][$word];
}
```
That do the job !
Hope that help

**Comment by markstory** [comment]
@atmacola That change causes tests to fail unfortunately:

```
1) InflectorTest::testPluralizeMultiWordIrregular
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'faq_categorias_preguntas_frecuentes'
+'faq_categoria_preguntas_frecuentes'
```

This was added to fix #6659

**Comment by markstory** [comment]
I wasn't able to reproduce this in the test harness in 4102961cb5e31f88f74fa4794ade7c6509dabc4c.
